### PR TITLE
Feat wishlist

### PR DESCRIPTION
**Pull request by humzahasan**
PR Description :

This PR contains file changes done for adding a generic Card component in app and using them on different section of homepage along with that router added to the project.

Use this link to see the files that needs to be reviewed
[Review Me Only](https://github.com/humzahasan/Hunger-Pangs/pull/3/files?file-filters%5B%5D=.css&file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.jsx&show-viewed-files=false)

### CODE REVIEW

File: src/pages/Homepage/Trendingitem/Trendingitem.jsx [hunk 1]
```diff
@@ -1,7 +1,47 @@
...
+    { name: "Salad", price: 100, productUrl: Salad },
+  ];
+
+  const addToCart = (item) => {
+    console.log("Add to cart", item);
+  };
+
+  const addToWishlist = (item) => {
+    console.log("Add to wishlist", item);
+  };
```
Comment (iRohitGaur) [inline src/pages/Homepage/Trendingitem/Trendingitem.jsx:24]: remove console.log from code

**Review by iRohitGaur — APPROVED** [review]
UI looks good. Nice Work. Good to merge 👍🏻

File: src/App.js [hunk 2]
```diff
@@ -1,10 +1,15 @@
 import "./App.css";
+import { Routes, Route } from "react-router-dom";
 import { Homepage } from "./pages/Homepage/Homepage";
+import { Wishlist } from "./pages/Wishlist/Wishlist";
```
Comment (sudipkundu999) [inline src/App.js:4]: Keep an `index.js` file in pages and export every page from there.
So when you'll actually import the pages they'll get imported as - 
`import {Homepage, Wishlist, Cart, Products, Login, Signup } from "./pages"`

**Review by sudipkundu999 — APPROVED** [review]
The code looks clean and easy to understand.
The PR description filter was really awesome.
This was something new to me.

Try to add an image of the component in the PR description
So that we get an idea of how it looks

**Review by zoef-yt — APPROVED** [review]
The UI is great. 
Code is understandable.
The PR description is great.
Do add images or preview links to the page.
Good to merge

File: src/pages/Homepage/Trendingitem/Trendingitem.jsx [hunk 3]
```diff
@@ -1,7 +1,47 @@
...
+
+  return (
+    <section className="homepage-trending">
+      <h2>Trending Items</h2>
+      <div className="trending-item">
+        {trendingItemList.map((item) => {
```
Comment (santhoshpemmaka) [inline src/pages/Homepage/Trendingitem/Trendingitem.jsx:30]: It's better to use one more condition inside jsx. Like {trendingItemsList && trendingItemsList.map()} using that avoid empty array using map statement. I'm using my codebase. Correct me if I'm wrong.

UI looking good 💯 . Good to Merge 👍

**Review by santhoshpemmaka — APPROVED** [review]

File: src/pages/Homepage/Trendingitem/Trendingitem.jsx (hunk 1 above)
Comment (humzahasan) [reply src/pages/Homepage/Trendingitem/Trendingitem.jsx:24]: > remove console.log from code

Will do!

File: src/App.js (hunk 2 above)
Comment (humzahasan) [reply src/App.js:4]: > Keep an `index.js` file in pages and export every page from there. So when you'll actually import the pages they'll get imported as - `import {Homepage, Wishlist, Cart, Products, Login, Signup } from "./pages"`

Great suggestion @sudipkundu999 ! Will do this 👍🏻

File: src/pages/Homepage/Trendingitem/Trendingitem.jsx (hunk 3 above)
Comment (humzahasan) [reply src/pages/Homepage/Trendingitem/Trendingitem.jsx:30]: > It's better to use one more condition inside jsx. Like {trendingItemsList && trendingItemsList.map()} using that avoid empty array using map statement. I'm using my codebase. Correct me if I'm wrong.
> 
> UI looking good 💯 . Good to Merge 👍

Good catch @santhoshpemmaka will check do this!

**Pull request merged**